src/network/networkd-link.c

```diff
@@ -1511,6 +1511,15 @@ static int link_up(Link *link) {
         }
 
         if (link->network->mtu) {
+                /* IPv6 protocol requires a minimum MTU of IPV6_MTU_MIN(1280) bytes
+                   on the interface. Bump up MTU bytes to IPV6_MTU_MIN. */
+                if (socket_ipv6_is_supported() && link->network->mtu < IPV6_MIN_MTU) {
```

Comment:
Good question. While I can see that people might want an explicit kill switch, I am tempted to say we should probably use an implicit one instead, given that we can probably say that any interface where no static IPv6, no DHCPv6 and no IPV6LL is on shall be considered an IPv6-less interface. That definition is simple enough I think (as long as it is clearly documented).

Thus, my suggestion would be to add a new helper function to the top if networkd-link.c, called link_ipv6_enabled(), that return false if socket_ipv6_is_supported() returns false, or when both of link_dhcp6_enabled() + link_ipv6ll_enabled() return false and if there's no static ipv6 address configured either. For the latter, it's probably a good idea to add a new function  to networkd-network.c or so, maybe called network_has_static_ipv6_addresses() or so, that iterates through the defined static addresses for the network until it finds an ipv6 one...

Then, let's set the disable_ipv6 sysctl whenever this function returns NULL on the specific interface. Most likely we can probably use the function in a couple of other cases too, where socket_ipv6_is_supported() is currently called.

(In theory, it would make sense to add the same for IPv4 too, but I figure the logic doesn't exist like that in the kernel for IPv4...)